test/demo/createCenter.inttest.ts

```diff
@@ -0,0 +1,204 @@
+        });
+    });
+
+    describe('When options does not include title and/or body', () => {
+        // Intentionally circumventing type check with cast for testing purposes
+        const options: NotificationOptions = {id: 'invalid-notification'} as NotificationOptions;
+
+        let createPromise: Promise<Notification>;
+        beforeEach(async () => {
+            createPromise = notifsRemote.create(testWindow.identity, options);
+
+            // We want to be sure the operation is completed, but don't care if it succeeds
+            await createPromise.catch(() => {});
+        });
+
+        afterEach(async () => {
+            // Cleanup the created notification (just in case it does get made)
+            await notifsRemote.clear(testWindow.identity, options.id!);
+        });
+
+        test('The promise rejects with a suitable error message', async () => {
+            await expect(createPromise).rejects.toThrow(/Invalid arguments passed to createNotification/);
+        });
+
+        test('A card is not added to the notification center', async () => {
+            const noteCards = await getCenterCardsByNotification(testApp.identity.uuid, options.id!);
+            expect(noteCards).toEqual([]);
+        });
+
+        test('The notification is not added to the persistence store', async () => {
+            const storedNotes = await getAppNotifications(testApp.identity.uuid);
+
+            expect(storedNotes).toEqual([]);
+        });
+
+        test('The notification does not appear in the result of getAll', async () => {
+            const appNotes = await notifsRemote.getAll(testWindow.identity);
+            expect(appNotes).toEqual([]);
+        });
+    });
+
+    describe('When options does not include id', () => {
+        const options: NotificationOptions = {body: 'Test Notification Body', title: 'Test Notificaiton Title'};
+
+        let createPromise: Promise<Notification>;
+        let note: Notification;
+        beforeEach(async () => {
+            createPromise = notifsRemote.create(testWindow.identity, options);
+
+            // We want to be sure the operation is completed, but don't care if it succeeds
+            note = await createPromise.catch(() => ({} as Notification));
+        });
+
+        test('The notification is created as expected', async () => {
+            await expect(createPromise).resolves;
+            expect(note).toMatchObject(options);
+
+            // Card is created
+            const noteCards = await getCenterCardsByNotification(testApp.identity.uuid, note.id);
+            expect(noteCards).toHaveLength(1);
+
+            // Card is correct
+            await assertDOMMatches(CardType.CENTER, testApp.identity.uuid, note);
+
+            // Notification is persisted
+            await assertNotificationStored(testWindow.identity, note);
+        });
+
+        test('An id is generated by the service and returned in the hydrated object', async () => {
```

Comment:
Using `ID` in comments and human text elsewhere

Reply:
There was a bit of a mix throughout. Changed all to ID so we're consistent.